main.go

```diff
@@ -92,14 +100,41 @@ type DNSRRValidator struct {
 	FailIfNotMatchesRegexp []string `yaml:"fail_if_not_matches_regexp"`
 }
 
-var Probers = map[string]func(string, http.ResponseWriter, Module) bool{
-	"http": probeHTTP,
-	"tcp":  probeTCP,
-	"icmp": probeICMP,
-	"dns":  probeDNS,
+var (
+	Probers = map[string]func(string, http.ResponseWriter, Module) bool{
+		"http": probeHTTP,
+		"tcp":  probeTCP,
+		"icmp": probeICMP,
+		"dns":  probeDNS,
+	}
+	sc = &SafeConfig{
+		C: &Config{},
+	}
+)
+
+func (this *SafeConfig) reloadConfig(confFile string) (err error) {
+	var c = Config{}
+
+	yamlFile, err := ioutil.ReadFile(confFile)
+	if err != nil {
+		log.Fatalf("Error reading config file: %s", err)
+		return err
+	}
+
+	this.Lock()
+	defer this.Unlock()
+
+	if err := yaml.Unmarshal(yamlFile, &c); err != nil {
+		log.Errorf("Error parsing config file: %s", err)
+		return err
+	}
+
+	*this.C = c
```

Comment:
`this.C = &c` would be faster, as you're copying a pointer rather than the entire struct.